examples/EDMF_data/uq_for_edmf.jl

```diff
@@ -0,0 +1,246 @@
+        inputs_constrained = reshape(
+            permutedims(input_constrained_mat, (2, 3, 1)),
+            (size(input_constrained_mat, 2), size(input_constrained_mat, 3) * size(input_constrained_mat, 1)),
+        )
+
+        outputs = reshape(
+            permutedims(output_mat, (2, 3, 1)),
+            (size(output_mat, 2), size(output_mat, 3) * size(output_mat, 1)),
+        )
+
+        # naive preprocessing of data
+        #remove extremes using truth cov
+        iqr_rows = mapslices(iqr, outputs, dims = 2)
+        med = mapslices(median, outputs, dims = 2)
+        twentyfive_rows = mapslices(x -> percentile(x, 25), outputs, dims = 2)
+        seventyfive_rows = mapslices(x -> percentile(x, 75), outputs, dims = 2)
+        max_bd = seventyfive_rows + 5 * iqr_rows
+        min_bd = twentyfive_rows - 5 * iqr_rows
+
+        idx_in_bd = (outputs .- min_bd) .> 0 .&& (max_bd .- outputs) .> 0
+        #remove all-bad particles
+        not_all_bad_particles = vec(any(idx_in_bd, dims = 1))
+        outputs = outputs[:, not_all_bad_particles]
+        inputs = inputs[:, not_all_bad_particles]
+        inputs_constrained = inputs_constrained[:, not_all_bad_particles]
+        println("Pruning all-bad particles...")
+        println("removed ", length(not_all_bad_particles)[1] - sum(not_all_bad_particles), " particles.")
+
+        #remove all-bad data dimensions
+        idx_in_bd = (outputs .- min_bd) .> 0 .&& (max_bd .- outputs) .> 0
+        not_all_bad_data_points = vec(any(idx_in_bd, dims = 2))
+        outputs = outputs[not_all_bad_data_points, :]
+        y_truth = y_truth[not_all_bad_data_points]
+        truth_cov = truth_cov[not_all_bad_data_points, not_all_bad_data_points]
+        println("Pruning all-bad data points...")
+        println("removed ", length(not_all_bad_data_points) - sum(not_all_bad_data_points), " data points.")
+
+        # quick plots of data
+        if PLOT_FLAG
+            println("plotting ensembles...")
+            for plot_i in 1:size(outputs, 1)
+                p = scatter(inputs_constrained[1, :], inputs_constrained[2, :], zcolor = outputs[plot_i, :])
+                savefig(p, joinpath(figure_save_directory, "output_" * string(plot_i) * ".png"))
+            end
+            println("finished plotting ensembles...")
+        end
+        input_output_pairs = DataContainers.PairedDataContainer(inputs, outputs)
+
+    end
+
+    # load and create prior distributions
+    prior_filepath = joinpath(exp_dir, "prior.jld2")
+    if !isfile(prior_filepath)
+        LoadError("prior file \"prior.jld2\" not found in directory \"" * exp_dir * "/\"")
+    else
+        prior_dict_raw = load(prior_filepath) #using JLD2
+        marginal_priors =
+            ParameterDistributions.ParameterDistribution.(
+                prior_dict_raw["distributions"],
+                prior_dict_raw["constraints"],
+                prior_dict_raw["u_names"],
+            )
```

Comment:
we will have to change this when we use new data (not now). I am now storing the ParameterDistribution object directly.